syncapi/storage/shared/syncserver.go

```diff
@@ -961,6 +1024,42 @@ func (d *Database) getStateDeltas(
 		return nil, nil, err
 	}
 
+	// find out which rooms this user is peeking, if any.
+	// We do this before joins so joins overwrite peeks
+	peeks, err := d.Peeks.SelectPeeks(ctx, userID, device.ID)
```

Comment:
This will be racey with `d.Peeks.MarkPeeksAsOld(ctx, txn, userID, device.ID)` - use the same `txn` please.